src/test/libraryfilesynctest.cpp

```diff
@@ -0,0 +1,712 @@
+                                trackRecordAfter.getSourceSynchronizedAt());
+            } else {
+                // Verify that the file has not been modified upon saving
+                ASSERT_EQ(fileLastModifiedAfter, fileLastModifiedBefore);
+                // Verify that the synchronization time stamp has not been updated
+                // if the file still exists.
+                EXPECT_TRUE(
+                        !fileLastModifiedAfter.isValid() ||
+                        trackRecordBefore.getSourceSynchronizedAt() ==
+                                trackRecordAfter.getSourceSynchronizedAt());
+            }
+        } else {
+            // Verify that the file has not been modified upon saving
+            ASSERT_EQ(fileLastModifiedAfter, fileLastModifiedBefore);
+            // Verify that the synchronization time stamp has not been updated.
+            EXPECT_EQ(
+                    trackRecordAfter.getSourceSynchronizedAt(),
+                    trackRecordBefore.getSourceSynchronizedAt());
+            if (importResult == mixxx::MetadataSource::ImportResult::Succeeded) {
+                // Verify that the metadata in the file has not been modified.
+                EXPECT_EQ(fileLastModifiedBefore, sourceSynchronizedAt);
+                EXPECT_EQ(trackRecordBefore.getMetadata(), importedTrackMetadata);
+            }
+        }
+
+        // Verify that the metadata in the database has been modified.
+        EXPECT_NE(trackRecordBefore.getMetadata(), trackRecordAfter.getMetadata());
+        EXPECT_EQ(newTitle, pTrack->getTitle());
+    }
+
+    QDateTime
+    fileLastModified() const {
+        return m_tempFileSystem.fileLastModified();
+    }
+
+    void updateFileLastModified() const {
+        m_tempFileSystem.updateFileLastModified();
+    }
+
+    void removeFile() const {
+        m_tempFileSystem.removeFile();
+    }
+
+    virtual TrackPointer prepareTestTrack() const = 0;
+
+    void checkTrackRecordSourceSyncStatus(
+            mixxx::TrackRecord::SourceSyncStatus expectedSourceSyncStatus) const {
+        const auto pTrack = prepareTestTrack();
+        const mixxx::TrackRecord emptyTrackRecord;
+        EXPECT_EQ(mixxx::TrackRecord::SourceSyncStatus::Void,
+                emptyTrackRecord.checkSourceSyncStatus(pTrack->getFileInfo()));
+        const auto trackRecordOfTestTrack = pTrack->getRecord();
+        EXPECT_EQ(expectedSourceSyncStatus,
+                trackRecordOfTestTrack.checkSourceSyncStatus(pTrack->getFileInfo()));
+    }
+
+  private:
+    void createTrack() {
+        const auto trackRef = TrackRef::fromFileInfo(m_tempFileSystem.fileInfo());
+        ASSERT_FALSE(trackCollectionManager()->getTrackByRef(trackRef));
+        const auto pTrack = getOrAddTrackByLocation(trackRef.getLocation());
+        ASSERT_TRUE(trackCollectionManager()->getTrackByRef(trackRef));
+        m_trackId = pTrack->getId();
+        sleepAfterFileLastModifiedUpdated();
```

Comment:
Why we need this delay? We already have such a delay when creating the file. So the track is already kind of aged.  
I think that should be moved into the tests to make it obvious why we need it.

Reply:
This one seems indeed redundant and a leftover from a previous rewrite. Will try.